src/cargo/core/compiler/compilation.rs

```diff
@@ -383,6 +383,11 @@ fn fill_rustc_tool_env(mut cmd: ProcessBuilder, unit: &Unit) -> ProcessBuilder {
         cmd.env("CARGO_BIN_NAME", name);
     }
     cmd.env("CARGO_CRATE_NAME", unit.target.crate_name());
+    let target_triple = match &unit.kind {
+        CompileKind::Host => host_triple,
+        CompileKind::Target(t) => t.short_name(),
```

Comment:
Should we use this instead so that local target JSON files are respected?

```suggestion
        CompileKind::Target(t) => t.rustc_target(),
```